### PR TITLE
[CC1101] Small fixes.

### PR DESCRIPTION
**Pull request by Guglio95**
Hi Jan,
I had some issues changing the sync word on my CC1101; 
I don't know why but setting it in burst mode was just applied to the SYNC1 register, SYNC0 was not affected. I put two [SPIsetRegValue](https://github.com/jgromes/RadioLib/compare/master...Guglio95:master#diff-bfbdc416744bf6d51bce6780aa1e9cbaR477). 
Then I fixed a wrong SYNC_MODE reg assignation.
Moreover i added a [reset on init](https://github.com/jgromes/RadioLib/compare/master...Guglio95:master#diff-bfbdc416744bf6d51bce6780aa1e9cbaR660) since the radio may still be configured by previous RadioLib instances.
Lastly I enabled the CRC check only if CRC validation is actually enabled.

Let me know.

### CODE REVIEW

**Comment by Guglio95** [comment]
Oops I just spot an issue:
[Here](https://github.com/jgromes/RadioLib/pull/90/commits/af9409b941e9419dec5d5f7e45dc961e51c6744a#diff-bfbdc416744bf6d51bce6780aa1e9cbaL456) the len parameter refers to the buffer size (I suppose).

[Here](https://github.com/jgromes/RadioLib/pull/90/commits/af9409b941e9419dec5d5f7e45dc961e51c6744a#diff-bfbdc416744bf6d51bce6780aa1e9cbaR593) the _syncWordLength refers to the fact that the " 2 bytes sync word" may be repeated obtaining a 4 bytes sync word....

Fun fact, how to handle this?

File: src/modules/CC1101/CC1101.cpp [hunk 1]
```diff
@@ -474,7 +474,8 @@ int16_t CC1101::setSyncWord(uint8_t* syncWord, uint8_t len, uint8_t maxErrBits)
   }
 
   // set sync word register
-  _mod->SPIwriteRegisterBurst(CC1101_REG_SYNC1, syncWord, len);
+  state = SPIsetRegValue(CC1101_REG_SYNC1, syncWord[0]);
+  state |= SPIsetRegValue(CC1101_REG_SYNC0, syncWord[1]);
```
Comment (jgromes) [inline src/modules/CC1101/CC1101.cpp]: Coudl you double check the byte order is correct? CC1101 datasheet says CC1101_REG_SYNC1 is the high byte.

File: src/modules/CC1101/CC1101.cpp [hunk 2]
```diff
@@ -474,7 +474,8 @@ int16_t CC1101::setSyncWord(uint8_t* syncWord, uint8_t len, uint8_t maxErrBits)
...
 
   return(ERR_NONE);
```
Comment (jgromes) [inline src/modules/CC1101/CC1101.cpp]: We should return `state` here.

**Comment by jgromes** [comment]
Do we need to support the case when sync word is repeated? I would argue not, but I don't really use CC1101. I only added it because it can do RTTY ;)

**Comment by Guglio95** [comment]
Hi Jan,
You are right, we may remove it! I’m not using the repeated sync word 😏

I’m off the grid, will fix it on Monday 👍

**Comment by Guglio95** [comment]
Hi Jan,

It should be ok now :+1: 
I found a wrong reg assignation (56cdbbd) 

Let me know.

**Comment by jgromes** [comment]
Merged, thanks!

**Pull request merged**